data/Store.js

```diff
@@ -122,9 +122,13 @@ export class Store {
         throwIf(!isEmpty(other), 'Unknown argument(s) passed to updateData().');
 
         // 1) Pre-process updates and adds into Records
+        const {summaryRecord} = this;
         let updateRecs, addRecs;
         if (update) {
-            updateRecs = update.map(it => this.createRecord(it));
+            updateRecs = update.map(it => {
+                const rec = it.id === summaryRecord?.id ? summaryRecord : this.getById(it.id);
```

Comment:
I might suggest getting `summaryRecord?.id` once before entering the loop - doubt it has a measurable impact, but still seems nicer to do the nullsafe lookup once and then compare to a local cost within the loop itself.

That said, should `getById()` instead be modified to always return the summaryRecord if there is one on the store and the asked-for ID is the summary ID?  Why not?

Reply:
I remember almost adding that to `getById()` but not sure why I decided against it, perhaps just to minimize the changes to commonly-used Store methods as I was originally thinking this would be a late merge into v28. Will go ahead and add it, can't think of any reason not to..